Automattic/wp-calypso
# Free Themes: Redirect to the Marketplace Thank You page

Fixes #73834

## Proposed Changes

* Check if a theme is free and from dotcom to redirect it to the Marketplace Thank You page
* Isolate logic to dispatch theme activation 

## Testing Instructions
* Go to the themes page (`/themes/:site`)
* Select a free theme and click on `Activate`
* You should see the Thanks modal on the Themes page
* Go to the themes page with the feature flag enabled (`/themes/:site?flags=themes/display-thank-you-page `)
* Select a free theme and click on `Activate`
* You should be redirected to the Marketplace Thank You Page

| Default version  | Feature flag enabled  |
| ------------- | ------------- |
|![CleanShot 2023-04-12 at 17 04 16@2x](https://user-images.githubusercontent.com/5039531/231572420-c6a7fe15-30c5-4e8d-ae12-ab7a4debea4a.png)|<img width="1274" alt="CleanShot 2023-04-12 at 17 05 08@2x" src="https://user-images.githubusercontent.com/5039531/231572565-077fc105-c3a4-48c5-bc00-db2f358f434d.png">|



## Pre-merge Checklist
